### PR TITLE
[FEATURE] Add variable assigning ViewHelper

### PR DESCRIPTION
**Pull request by NamelessCoder**
New ViewHelper `f:variable` with arguments "name"
and "value", assigns a variable in the VariableProvider
using the `add()` method and does not remove the
variable again afterwards.

Usages:

    {f:variable(name: 'myvariable', value: 'some value')}
    <f:variable name="myvariable">some value</f:variable>
    {oldvariable -> f:format.htmlspecialchars() -> f:variable(name: 'newvariable')}
    <f:variable name="myvariable"><f:format.htmlspecialchars>{oldvariable}</f:format.htmlspecialchars></f:variable>

### CODE REVIEW

File: src/ViewHelpers/VariableViewHelper.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ *     <f:variable name="myvariable">some value</f:variable>
+ *     {oldvariable -> f:format.htmlspecialchars() -> f:variable(name: 'newvariable')}
+ *     <f:variable name="myvariable"><f:format.htmlspecialchars>{oldvariable}</f:format.htmlspecialchars></f:variable>
+ *
+ * Please note:
+ *
+ * If you assign a variable with a dotted name, e.g.
+ * "myarray.property" then this may or may not be
+ * supported by the VariableProvider that's being
+ * used. The default
```
Comment (mneuhaus) [inline src/ViewHelpers/VariableViewHelper.php]: The default...?

Comment (mneuhaus) [inline src/ViewHelpers/VariableViewHelper.php]: + i'm not really sure we should allow a dot in the name. My gut tells me, that prohibiting dot here will save use plenty of headaches later on

Comment (NamelessCoder) [inline src/ViewHelpers/VariableViewHelper.php]: That part can go in another PR after we discuss it ;)
